pymeasure/instruments/instrument.py

```diff
@@ -114,14 +155,24 @@ def binary_values(self, command, header_bytes=0, dtype=np.float32):
         return self.adapter.binary_values(command, header_bytes, dtype)
 
     @staticmethod
-    def control(get_command, set_command, docs,
-                validator=lambda v, vs: v, values=(), map_values=False,
-                get_process=lambda v: v, set_process=lambda v: v,
-                check_set_errors=False, check_get_errors=False,
-                **kwargs):
+    def _general_control(get_command,
+                         set_command,
+                         docs,
+                         validator=lambda v, vs: v,
+                         values=(),
+                         map_values=False,
+                         get_process=lambda v: v,
+                         set_process=lambda v: v,
+                         command_process=lambda c: c,
```

Comment:
Well, `command_process` was present in the existing implementation of `measurement` with the following comment:
```
        :param command_process: A function that take a command and allows processing
                            before executing the command, for both getting and setting

```
I left it to avoid breaking anything already existing and relying on such parameter. I noticed also that, for some reason, it was present only in the `measurement` method and so I decided to fix the comment. I have no problem in removing it, but this was not added by me.

Reply:
So this is kinda like our `preprocess_reply`, but for the sent commands? I'm not sure why this was here in the first place. As it's an inconsistency in our API (only in `measurement`) and I can't find any usage of it in the codebase at least, we should consider removing it (or at least deprecating it and removing then in 1.0)

Reply:
Second thought: I would consider this as an extra flexibility which may be useful, but I am not 100% sure.
I would propose to file an issue, whether we want to keep or remove it and have some feedback from the community, so it is not forgotten and can possibly addressed in another PR.

Reply:
> Second thought: I would consider this as an extra flexibility which may be useful, but I am not 100% sure.
I would propose to file an issue, whether we want to keep or remove it and have some feedback from the community, so it is not forgotten and can possibly addressed in another PR.

Agreed, this sounds like a good resolution. Do you have time to take care of this?

Reply:
Sure, I will do this. Thanks.